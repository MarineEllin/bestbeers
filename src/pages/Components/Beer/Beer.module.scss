.beerCard {
  display: flex;
  flex-direction: column;
  align-items: center;
  max-width: 350px;
  .imageContainer {
    border-top-left-radius: var(--border-radius);
    border-top-right-radius: var(--border-radius);
    width: 100%;
    height: 250px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: var(--light);
    .image {
      height: 150px;
      transition: scale 1s;
      &:hover {
        scale: 1.3;
      }
    }
  }
  .infoContainer {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    border-bottom: 1px solid var(--light);
    border-right: 1px solid var(--light);
    border-left: 1px solid var(--light);
    border-bottom-left-radius: var(--border-radius);
    border-bottom-right-radius: var(--border-radius);
    width: 100%;
    height: 350px;
    padding: 20px;
    .name {
      color: var(--primary-light);
      font-weight: 700;
    }
    .description {
      display: -webkit-box;
      text-overflow: ellipsis;
      overflow: hidden;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 3;
      font-size: small;
      line-height: 18px;
    }
    .food {
      font-size: small;
      height: 100px;
      ul {
        padding: 0;
        line-height: 18px;
      }
    }
    .indicators {
      width: 100%;
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: space-between;
      font-size: small;
      height: 40px;

      .bitternessBorderBar {
        border: 1px solid var(--primary-light);
        border-radius: var(--border-radius);
        height: 10px;
        box-shadow: var(--box-shadow-indicators);
        .bitternessBar {
          background-color: var(--primary-light);
          height: 9px;
          border-radius: var(--border-radius);
        }
      }

      .alcoholContainer {
        background-color: var(--primary-light);
        width: 45px;
        height: 45px;
        display: flex;
        justify-content: center;
        align-items: center;
        border-radius: 50%;
        box-shadow: var(--box-shadow-indicators);
      }
    }
  }
}
