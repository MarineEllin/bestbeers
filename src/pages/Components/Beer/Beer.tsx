import { Constants } from "constants/constants";
import styles from "./Beer.module.scss";
import { BeerObject } from "models/beer.model";

function Beer({ beer }: { beer: BeerObject }) {
  return (
    <div className={styles.beerCard}>
      <div className={styles.imageContainer}>
        <img src={beer.image_url} alt="" className={styles.image} />
      </div>
      <div className={styles.infoContainer}>
        <div>
          <div className={styles.name}>{beer.name}</div>
          <div className={styles.descriptionContainer}>
            <div className={styles.description}>{beer.description}</div>
          </div>

          <div className={styles.food}>
            <ul>
              {beer.food_pairing.map((food, index) => (
                <li key={index}>
                  <i className="fa-solid fa-utensils mr-5 font-color-primary-light" />
                  {food}
                </li>
              ))}
            </ul>
          </div>
        </div>

        <div className={styles.indicators}>
          <div>
            <div className="mb-5">Bitterness</div>
            <div
              className={styles.bitternessBorderBar}
              style={{
                width: Constants.MAX_IBU_INDICATOR,
              }}
            >
              <div
                className={styles.bitternessBar}
                style={{
                  width: beer.ibu != null ? beer.ibu : 0,
                }}
              ></div>
            </div>
          </div>
          <div className={styles.alcoholContainer}>
            <div>{beer.abv}%</div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Beer;
