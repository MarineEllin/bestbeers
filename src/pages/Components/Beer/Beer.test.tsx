import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Beer from "./Beer";
import { RecoilRoot } from "recoil";

test("Write a test for Header with snapshot", () => {
  const beerObj = {
    id: 0,
    name: "my beer",
    description: "my beer description",
    image_url: "my beer image url",
    abv: 7,
    ibu: 30,
    food_pairing: ["fish", "chocolate"],
  };
  const component = renderer.create(
    <>
      <RecoilRoot>
        <Beer beer={beerObj} />
      </RecoilRoot>
      <Beer beer={beerObj} />
    </>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Write a test for Beer with DOM", async () => {
  // eslint-disable-next-line testing-library/render-result-naming-convention
  const beerObj = {
    id: 0,
    name: "my beer",
    description: "my beer description",
    image_url: "my beer image url",
    abv: 7,
    ibu: 30,
    food_pairing: ["fish", "chocolate"],
  };
  const view = render(
    <RecoilRoot>
      <Beer beer={beerObj} />
    </RecoilRoot>
  );
  // eslint-disable-next-line testing-library/prefer-screen-queries
  const haveAnElWithAlText = view.getByText(/Bitterness/);
  expect(haveAnElWithAlText).toBeTruthy();
});
