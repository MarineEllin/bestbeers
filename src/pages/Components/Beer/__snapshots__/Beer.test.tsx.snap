// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Write a test for Header with snapshot 1`] = `
[
  <div
    className="beerCard"
  >
    <div
      className="imageContainer"
    >
      <img
        className="image"
        src="my beer image url"
      />
    </div>
    <div
      className="infoContainer"
    >
      <div>
        <div
          className="name"
        >
          my beer
        </div>
        <div
          className="descriptionContainer"
        >
          <div
            className="description"
          >
            my beer description
          </div>
        </div>
        <div
          className="food"
        >
          <ul>
            <li>
              <i
                className="fa-solid fa-utensils mr-5 font-color-primary-light"
              />
              fish
            </li>
            <li>
              <i
                className="fa-solid fa-utensils mr-5 font-color-primary-light"
              />
              chocolate
            </li>
          </ul>
        </div>
      </div>
      <div
        className="indicators"
      >
        <div>
          <div
            className="mb-5"
          >
            Bitterness
          </div>
          <div
            className="bitternessBorderBar"
            style={
              {
                "width": 100,
              }
            }
          >
            <div
              className="bitternessBar"
              style={
                {
                  "width": 30,
                }
              }
            />
          </div>
        </div>
        <div
          className="alcoholContainer"
        >
          <div>
            7
            %
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="beerCard"
  >
    <div
      className="imageContainer"
    >
      <img
        className="image"
        src="my beer image url"
      />
    </div>
    <div
      className="infoContainer"
    >
      <div>
        <div
          className="name"
        >
          my beer
        </div>
        <div
          className="descriptionContainer"
        >
          <div
            className="description"
          >
            my beer description
          </div>
        </div>
        <div
          className="food"
        >
          <ul>
            <li>
              <i
                className="fa-solid fa-utensils mr-5 font-color-primary-light"
              />
              fish
            </li>
            <li>
              <i
                className="fa-solid fa-utensils mr-5 font-color-primary-light"
              />
              chocolate
            </li>
          </ul>
        </div>
      </div>
      <div
        className="indicators"
      >
        <div>
          <div
            className="mb-5"
          >
            Bitterness
          </div>
          <div
            className="bitternessBorderBar"
            style={
              {
                "width": 100,
              }
            }
          >
            <div
              className="bitternessBar"
              style={
                {
                  "width": 30,
                }
              }
            />
          </div>
        </div>
        <div
          className="alcoholContainer"
        >
          <div>
            7
            %
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
